gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -219,7 +229,8 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
    * @throws IOException
    */
   @VisibleForTesting
-  static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobStatus, StateStore stateStore)
+  static void addJobStatusToStateStore(org.apache.gobblin.configuration.State jobStatus, StateStore stateStore,
+      Optional<GaaSObservabilityEventProducer> eventProducer)
```

Comment:
Curious on our philosophy  for using optional as a parameter. Why not use nulls + overloading. Java optionals are best used as a return value but as an input, I think it makes the developer experience less readable (There are plenty of debates on it on the internet). 

Although, I have seen the pattern used in some other parts of Gobblin, so clearly past developers felt it was okay.

Reply:
I've not heard that debate....

overall, the idea w/ `Optional` is to encode possible absence in the type system where it can be checked statically and automatically.  `null` is notoriously furtive... take for instance this very impl. (here), which does not check `stateStore != null`--but should it?  the practice I follow is that when that ought to be checked, the expectation should be stated, not in javadoc or other code comments, but via `Optional`.

again, I've not heard specific concern w/ use as a parameter type.  quite the opposite, in fact: the use is noted to bring clear benefit.

Reply:
I don't have a strong preference for one or the other as long as the team agrees on a logically consistent pattern. This topic has come up in my past roles, so I figured I'd ask out of curiosity. Good points about the API being more self descriptive as opposed to an annotation like `@Nullable` which is more hidden

The argument against `Optional` as as follows:
- WIthout optional, you can have 2 possible states `null` or actual reference
- With optional, you can have 3 possible states `null`, `Optional.empty()`, `Optional.present()`

It's reasonable to try your best not to pass null as a parameter. (e.g. Optional.ofNullable(...)). But it does add visual clutter from a readability perspective that doesn't add much value when method overloading would be totally sufficient. 

This is especially true when there is only one optional parameter. Here, we don't have a bunch of permutations of options that the method caller could use (Which is probably a code smell on its own). 

Some debates in case you want to go down the rabbit hole / get nerd sniped:
https://rules.sonarsource.com/java/RSPEC-3553
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments
http://dolszewski.com/java/java-8-optional-use-cases/
https://stackoverflow.com/questions/26327957/should-java-8-getters-return-optional-type/26328555#26328555

https://docs.oracle.com/javase/10/docs/api/java/util/Optional.html
```
Optional is primarily intended for use as a method return type where there is a clear need to represent "no result," and where using null is likely to cause errors. A variable whose type is Optional should never itself be null; it should always point to an Optional instance.
```

Reply:
I think I'm going to lean towards going neither down the route of nulls or Optionals, and make a NoopEventSubmitter which will act as the default class from the factory, and its behavior is to do no work.